File: PHP/MaximumErasureValueTest.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit MaximumErasureValueTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("MaximumErasureValue.php");

class MaximumErasureValueTest extends TestCase {

    public function testMaximumErasureValue() {
        $obj = new MaximumErasureValue();
        //Test 1
        $want = 17;
        $got = $obj->maximumUniqueSubarray( [4,2,4,5,6] );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = 8;
        $got = $obj->maximumUniqueSubarray( [5,2,1,2,5,2,1,2,5] );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/MaximumErasureValue_test.go
/**
 * @version GO 1.23.5
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestMaximumErasureValue
 */
package GO

import (
	"reflect"
	"testing"
)

func TestMaximumErasureValue(t *testing.T) {
	tables := []struct {
		want int
		nums []int
	}{
		{17, []int{4, 2, 4, 5, 6}},
		{8, []int{5, 2, 1, 2, 5, 2, 1, 2, 5}},
	}

	for _, table := range tables {
		got := maximumUniqueSubarray(table.nums)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
